### PR TITLE
[pcc-437] Add components command and smart component map validation helpers.

### PR DESCRIPTION
**Pull request by kevinstubbs**

### CODE REVIEW

File: packages/cli/src/cli/commands/sites.ts [hunk 1]
```diff
@@ -16,6 +18,39 @@ export const createSite = errorHandler<string>(async (url: string) => {
   }
 });
 
+type getComponentSchemaParams = { url: string; apiPath: string | null };
+export const getComponentSchema = errorHandler<getComponentSchemaParams>(
+  async ({ url, apiPath }: getComponentSchemaParams) => {
+    const spinner = ora("Creating site...").start();
```
Comment (Aumkar) [inline packages/cli/src/cli/commands/sites.ts:24]: I think it should be `Retrieving component schema`

File: packages/cli/src/cli/commands/sites.ts [hunk 2]
```diff
@@ -16,6 +18,39 @@ export const createSite = errorHandler<string>(async (url: string) => {
...
+    try {
```
Comment (Aumkar) [inline packages/cli/src/cli/commands/sites.ts:25]: If its to handle unexpected errors, you can remove it.
`errorHandler` will take care of it and also display stac trace.

File: packages/cli/src/cli/commands/sites.ts [hunk 3]
```diff
@@ -16,6 +18,39 @@ export const createSite = errorHandler<string>(async (url: string) => {
...
+      }`;
+      const result = (await axios.get(schemaEndpoint)).data;
+
+      spinner.succeed(
+        `Retrieved component schema from ${schemaEndpoint}. Now checking its validity`,
+      );
+
+      try {
+        validateComponentSchema(result);
+      } catch (e) {
```
Comment (Aumkar) [inline packages/cli/src/cli/commands/sites.ts:37]: May be we can 
```
console.error(chalk.red(<message>))
process.exit(1)
```
Just to be consistent with other error message formatting.

Comment (kevinstubbs) [reply packages/cli/src/cli/commands/sites.ts:37]: Do we need to call spinner.fail() too?

**Pull request merged**